connector/src/main/java/org/jboss/as/connector/util/CopyOnWriteArrayListMultiMap.java

```diff
@@ -18,42 +18,31 @@ public List<V> get(K k) {
         return cache.get(k);
     }
 
-    public synchronized List<V> remove(K k) {
+    public List<V> remove(K k) {
         return cache.remove(k);
     }
 
-    public synchronized void putIfAbsent(K k, V v) {
+    public void putIfAbsent(K k, V v) {
```

Comment:
This will be prone to lost updates, e.g. for concurrent calls to putIfAbsent(...) for the same key, where no map entry yet exists.  Rather, I think you want something like:
```java
cache.computeIfAbsent(k, key -> new CopyOnWriteArrayList<>()).addIfAbsent(v);
```

Reply:
fixed